File: core/src/main/java/com/gavin/core/application/BaseAppLogic.java
package com.gavin.core.application;

import android.content.res.Configuration;
import androidx.annotation.NonNull;

/**
 * @author : com.gavin
 * @date 2018/8/28.
 */
public class BaseAppLogic {
    protected static BaseApplication mApplication;
    public BaseAppLogic() {}

    public void setApplication(@NonNull BaseApplication application) {
        mApplication = application;
    }

    public static BaseApplication getApplication() {
        return mApplication;
    }

    public void onCreate(){}
    public void onTerminate(){}
    public void onLowMemory(){}
    public void onTrimMemory(int level){}
    public void onConfigurationChanged(Configuration newConfig){}
}

File: core/src/main/java/com/gavin/core/application/BaseAppLogic.kt
package com.gavin.core.application

import android.content.res.Configuration

/**
 * @author : com.gavin
 * @date 2018/8/28.
 */
open class BaseAppLogic {

    companion object {
        @JvmField
        var mApplication: BaseApplication? =null

        @JvmStatic
        fun getApplication():BaseApplication{
            return mApplication?:throw NullPointerException("must setApplication before")
        }
    }

    open fun onCreate() {}
    open fun onTerminate() {}
    open fun onLowMemory() {}
    open fun onTrimMemory(level: Int) {}
    open fun onConfigurationChanged(newConfig: Configuration) {}

    fun setApplication(application:BaseApplication){
        mApplication = application
    }

}
